pybedlite/overlap_detector.py

```diff
@@ -351,27 +371,15 @@ def get_overlaps(self, interval: Span) -> List[SpanType]:
                 * The interval's strand, positive or negative (assumed to be positive if undefined)
                 * The interval's reference sequence name (lexicographically)
         """
-        tree = self._refname_to_tree.get(interval.refname)
-        if tree is None:
-            return []
-        else:
-            if not self._refname_to_indexed[interval.refname]:
-                tree.index()
-            ref_intervals: List[SpanType] = self._refname_to_intervals[interval.refname]
-            # NB: only return unique instances of intervals
-            intervals: Set[SpanType] = {
-                ref_intervals[index]
-                for _, _, index in tree.overlap(interval.refname, interval.start, interval.end)
-            }
-            return sorted(
-                intervals,
-                key=lambda intv: (
-                    intv.start,
-                    intv.end,
-                    self._negative(intv),
-                    intv.refname,
-                ),
-            )
+        return sorted(
+            set(self.iter_overlaps(interval)),
```

Comment:
Curious about the `set` here - how does `IntervalSet` handle duplicate intervals?

Reply:
`IntervalSet` will only be storing indices into the intervals list that the `OverlapDetector` stores. So it won't even be aware of any _exact_ duplicates (i.e. even the index is the same). As for intervals with duplicate start and stop positions, it stores them correctly.

Reply:
It will depend on the hash of the object you place in the `OverlapDetector`, since the overlap detector is generic now.

If you send in a dataclass with `frozen=True`, then all fields will be used as a part of the hash. However, if you make a custom interval-like object and define your own hash method, then that hash method will be used.

Both [`BedRecord`](https://github.com/fulcrumgenomics/pybedlite/blob/a63c4926f86607930dd0a58aeb75523e79a350f0/pybedlite/bed_record.py#L45) and [`Interval`](https://github.com/fulcrumgenomics/pybedlite/blob/a63c4926f86607930dd0a58aeb75523e79a350f0/pybedlite/overlap_detector.py#L114) hash on all their fields.

Mypy should disallow any custom interval-like object that does not have a `__hash__()` dunder method:

https://github.com/fulcrumgenomics/pybedlite/blob/a63c4926f86607930dd0a58aeb75523e79a350f0/pybedlite/overlap_detector.py#L82

I don't think a call to `set` or `sorted` should have been included in the original implementation (a user should have the choice of "unique-ifying" later, or sorting later) but because they are already a part of the implementation details, it makes sense to preserve behavior:

https://github.com/fulcrumgenomics/pybedlite/blob/9c4990edda7d2a0687e33d033a226a38ba30bec4/pybedlite/overlap_detector.py#L362-L374